integrationtests/gitjob/controller/controller_test.go

```diff
@@ -226,6 +251,112 @@ var _ = Describe("GitJob controller", func() {
 			}).Should(BeTrue())
 		})
 	})
+
+	When("creating a gitRepo that references a non existing helm secrets", func() {
+		var (
+			gitRepo                v1alpha1.GitRepo
+			gitRepoName            string
+			helmSecretNameForPaths string
+			helmSecretName         string
+		)
+
+		JustBeforeEach(func() {
+			gitRepoName = "test-no-for-paths-secret"
+			gitRepo = createGitRepo(gitRepoName)
+			gitRepo.Spec.HelmSecretNameForPaths = helmSecretNameForPaths
+			gitRepo.Spec.HelmSecretName = helmSecretName
+			// Create should return an error
+			err := k8sClient.Create(ctx, &gitRepo)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(simulateGitPollerUpdatingCommitInStatus(gitRepo, commit)).ToNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			err := k8sClient.Delete(ctx, &gitRepo)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		Context("helmSecretNameForPaths secret does not exist", func() {
+			BeforeEach(func() {
+				helmSecretNameForPaths = "secret-does-not-exist"
+				helmSecretName = ""
+			})
+			It("logs an error about HelmSecretNameForPaths not being found", func() {
+				Eventually(func() bool {
+					strLogs := logsBuffer.String()
+					return strings.Contains(strLogs, "failed to look up HelmSecretNameForPaths, error: Secret \\\"secret-does-not-exist\\\" not found")
```

Comment:
nit: this would probably also work (same remark about L327):
```suggestion
					return strings.Contains(strLogs, `failed to look up HelmSecretNameForPaths, error: Secret \"secret-does-not-exist\" not found`)
```

Reply:
changed as suggested